Description: Disable WXDEBUG by default
 wxWidgets 3.0 turns on WXDEBUG assertions by default, whereas 2.8
 quietly dealt with them.  For C++ apps, you can build with -DNDEBUG or
 use wxDisableAsserts() at run-time, but wxPython doesn't provide a
 way to turn them off.  So for now we turn them off by default in the
 Debian wxpython3.0 packages, and people can enable them by setting
 environment variable DEBIAN_WXPYTHON_ENABLE_WXDEBUG to a non-empty value.
Author: Olly Betts <olly@survex.com>
Forwarded: no
Last-Update: 2014-08-18

diff --git a/wxPython/src/helpers.cpp b/wxPython/src/helpers.cpp
index e3d13c9..f945073 100644
--- a/wxPython/src/helpers.cpp
+++ b/wxPython/src/helpers.cpp
@@ -47,6 +47,8 @@
 #include <wx/mimetype.h>
 #include <wx/image.h>
 
+#include <stdlib.h>
+
 //----------------------------------------------------------------------
 
 #if PYTHON_API_VERSION < 1009 && wxUSE_UNICODE
@@ -746,6 +748,18 @@ void __wxPyPreStart(PyObject* moduleDict)
     wxApp::CheckBuildOptions(WX_BUILD_OPTIONS_SIGNATURE, "wxPython");
 
     wxInitAllImageHandlers();
+
+    // wxWidgets 3.0 turns on WXDEBUG assertions by default, whereas 2.8
+    // quietly dealt with them.  For C++ apps, you can build with -DNDEBUG or
+    // use wxDisableAsserts() at run-time, but wxPython doesn't provide a
+    // way to turn them off.  So for now we turn them off by default in the
+    // Debian wxpython3.0 packages, and people can enable them by setting
+    // environment variable DEBIAN_WXPYTHON_ENABLE_WXDEBUG to a non-empty
+    // value.
+    const char * p = getenv("DEBIAN_WXPYTHON_ENABLE_WXDEBUG");
+    if (!p || !*p) {
+        wxDisableAsserts();
+    }
 }
 
 
